application/src/main/scala/com/azavea/franklin/api/endpoints/SearchEndpoints.scala

```diff
@@ -39,12 +45,20 @@ object SearchEndpoints {
             collections getOrElse Nil,
             ids getOrElse Nil,
             limit,
-            next,
-            Map.empty
+            Map.empty,
+            token
           )
         }
       )(sf => (sf.datetime, sf.bbox, Some(sf.collections), Some(sf.items), sf.limit, sf.next))
 
+  val searchPostInput: EndpointInput[SearchFilters] =
+    jsonBody[SearchFilters]
+      .and(nextToken)
+      .map[SearchFilters]({ (tup: (SearchFilters, Option[PaginationToken])) =>
+        val (filters, token) = tup
+        filters.copy(next = filters.next orElse token)
```

Comment:
This could arguably always be replacement with the query param token -- I didn't do that because I'm counting on no one having access to the information they'd need to construct the non-opaque version and just using the information that comes back in the search result `links` field to find their way around (and that `body` always has `next` set to `null`).